Main/SEToolbox/SEToolbox/Models/WorldResource.cs

```diff
@@ -356,22 +357,24 @@ public XmlDocument LoadSectorXml()
 
             try
             {
-                if (ZipTools.IsGzipedFile(filename))
-                {
-                    // New file format is compressed.
-                    // These steps could probably be combined, but would have to use a MemoryStream, which has memory limits before it causes performance issues when chunking memory.
-                    // Using a temporary file in this situation has less performance issues as it's moved straight to disk.
-                    var tempFilename = TempfileUtil.NewFilename();
-                    ZipTools.GZipUncompress(filename, tempFilename);
-                    xDoc.Load(tempFilename);
-                    _compressedCheckpointFormat = true;
-                }
-                else
-                {
-                    // Old file format is raw XML.
-                    xDoc.Load(filename);
-                    _compressedCheckpointFormat = false;
-                }
+	            using ( FileStream sectorFile = File.OpenRead( filename ) )
+	            {
+		            if (sectorFile.CheckGZipHeader(  ))
+		            {
+			            using ( GZipStream zippedFile = new GZipStream( sectorFile, CompressionMode.Decompress ) )
+			            {
+				            xDoc.Load(zippedFile);
```

Comment:
I assume it's this part you're worried about?

Reply:
Still going to test with a large world, just to be sure.

Reply:
...Hm...Actually, the DeflateStream source looks like it might have to re-inflate the entire stream all the way to the point requested, on each call to Read, which would make the XmlDocument::Load() function be pretty damn inefficient, if true.
Oh well. Can still extract to file and then read from there to avoid that, using the pure CLR classes.

Reply:
Would it be any better to use the xDoc.Load(Stream inStream) ?

Reply:
I'd have to look at that code, too.
I'll have some time to take a look in a few hours (in the evening, EST).
My bet, though, is that it also makes calls to Read() in a loop, which would have the same issue, since the problem lies in DeflateStream's implementation of Read().

Reply:
Didn't get to it yet (obviously) but it's on my list. Probably get to it this weekend.